### PR TITLE
python3Packages.starlette: 0.13.8 -> 0.14.2

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

https://github.com/encode/starlette/releases/tag/0.14.0
https://github.com/encode/starlette/releases/tag/0.14.1
https://github.com/encode/starlette/releases/tag/0.14.2

- ujson was dropped in https://github.com/encode/starlette/pull/1047
  - removal of `UJSONResponse` breaks fastapi https://github.com/tiangolo/fastapi/pull/2335
- test_debug_html was fixed in https://github.com/encode/starlette/pull/1132

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112701` at 8732d1b7 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.starlette</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/644fb155bcb9b9f3b3a9bfba6c8279ab">python37Packages.fastapi</a></li></li>
    <li><a href="https://gist.github.com/63285c8488ca4b639313e292599b354b">python38Packages.fastapi</a></li></li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/8e18eafd7b92b0f5982f029d45162b97">python37Packages.starlette</a></li></li>
    <li><a href="https://gist.github.com/315429e7fbf586fd323bb05bd97287e4">python38Packages.starlette</a></li></li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112701` at 8732d1b7 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.starlette</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/a96dc57a659fab9d97d0e50b9ed52236">python37Packages.fastapi</a></li></li>
    <li><a href="https://gist.github.com/bb26fe29133525685459c2beaa3194f2">python38Packages.fastapi</a></li></li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/acd54d994c66b87a20aac1092c453eb2">python37Packages.starlette</a></li></li>
    <li><a href="https://gist.github.com/9f57ff44316cfba841e5ad5e4c8821f7">python38Packages.starlette</a></li></li>
  </ul>
</details>

**Comment by mweinelt** [comment]
Rebased, because nixpkgs-review by @r-rmcgibbo didn't take the uvloop fixes on the target branch into account.

Result of `nixpkgs-review pr 112701` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python37Packages.fastapi</li>
    <li>python38Packages.fastapi</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.starlette</li>
    <li>python38Packages.starlette</li>
    <li>python39Packages.starlette</li>
  </ul>
</details>

**Comment by rmcgibbo** [comment]
Sorry about that

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 112701` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python37Packages.fastapi</li>
    <li>python38Packages.fastapi</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.starlette</li>
    <li>python38Packages.starlette</li>
    <li>python39Packages.starlette</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package build (see https://github.com/NixOS/nixpkgs/pull/112701#issuecomment-777249617)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-112701/results/python38Packages.starlette/lib/python3.8/site-packages]$ python
Python 3.7.9 (default, Aug 15 2020, 06:36:35) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import starlette
>>> starlette.__version__
'0.14.2'
...ew/pr-112701/results/python38Packages.starlette/lib/python3.8/site-packages]$ python example.py
INFO:     Started server process [4522]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:5000 (Press CTRL+C to quit)
```

Looks good to me.

**Comment by mweinelt** [comment]
> Sorry about that

Not actually your fault, the fixes arrived on master three minutes after this pull request was opened. I was merely stating the fact, that they didn't (couldn't) take that into account.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112701` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.fastapi: log <a href="https://termbin.com/7syp">https://termbin.com/7syp</a></li>
<li>python38Packages.fastapi: log <a href="https://termbin.com/o88x">https://termbin.com/o88x</a></li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.starlette</li>
<li>python38Packages.starlette</li>
<li>python39Packages.starlette</li>
</ul>
</details>

**Comment by mweinelt** [comment]
Because fastapi is the only consumer of starlette, I'd say we wait for feedback from fastapi maintainers.

**Comment by oxzi** [comment]
The latest `platformio` release requires a `starlette` version of `0.14.x`. Thus, I'd love to this one merged.

**Comment by wd15** [comment]
It seems sensible to disable FastAPI as it has been stuck at `0.13.x` of Starlette for quite some time. FastAPI also fixes the exact Starlette bug fix version, which doesn't work well with Nixpkgs. There is an open [FastAPI PR](https://github.com/tiangolo/fastapi/pull/2335) that hasn't been reviewed since December. So, I'm in favor of disabling FastAPI and updating Starlette especially if Starlette's dependencies are blocked.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 112701` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python38Packages.fastapi</li>
    <li>python39Packages.fastapi</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python38Packages.starlette</li>
    <li>python39Packages.starlette</li>
  </ul>
</details>

**Pull request merged**
